py/server/deephaven/agg.py

```diff
@@ -278,33 +280,35 @@ def std(cols: Union[str, List[str]] = None) -> Aggregation:
     return Aggregation(j_agg_spec=_JAggSpec.std(), cols=cols)
 
 
-def unique(cols: Union[str, List[str]] = None, include_nulls: bool = False, null_sentinel: Any = None) -> Aggregation:
-    """Create a Unique aggregation which computes the single unique value within an aggregation group for each of
+def unique(cols: Union[str, List[str]] = None, include_nulls: bool = False, non_unique_sentinel: Any = None) -> Aggregation:
+    """Creates a Unique aggregation which computes the single unique value within an aggregation group for each of
     the given columns. If all values in a column are null, or if there is more than one distinct value in a column, the
-    result is null.
+    result is null or the specified non_unique_sentinel value.
 
     Args:
         cols (Union[str, List[str]]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
             default is None, only valid when used in Table agg_all_by operation
         include_nulls (bool): whether null is treated as a value for the purpose of determining if the values in the
             aggregation group are unique, default is False.
-        null_sentinel (Any): the non-null sentinel value to substitute null when include_nulls is True, default is None
+        non_unique_sentinel (Any): the non-null sentinel value when no unique value exists, default is None, must be
+            a non-None value when include_nulls is True.
 
     Returns:
         an aggregation
     """
-    if include_nulls:
-        if null_sentinel:
-            null_sentinel = _JUnionObject.of(null_sentinel)
-        else:
+
+    if non_unique_sentinel is None:
+        if include_nulls:
```

Comment:
I don't think we actually impose this limitation in the Java code. I suppose in your case it's reasonable, since the argument is optional.